devito/passes/iet/misc.py

```diff
@@ -57,13 +57,30 @@ def hoist_prodders(iet):
 @iet_pass
 def relax_incr_dimensions(iet, **kwargs):
     """
-    Recast Iterations over IncrDimensions as ElementalFunctions; insert
-    ElementalCalls to iterate over the "main" and "remainder" regions induced
-    by the IncrDimensions.
+    This pass adjusts the bounds of blocked loops in order to
+    include the "remainder regions". The Iterations in the input IET span
+    only up to the end of the last of the blocks that fully fit in the domain,
+    thus missing the "remainder" part.
+
+    Without the relaxation that occurs in this pass, the only way to iterate over the
+    entire iteration space is to have step increments/ block shapes that are perfect
+    divisors of the iteration space.
+
+    e.g. In case of an iteration space of size 67 and block size 8
+    only 64 iterations would be computed as `67 - 67mod8 = 64`
+
+    A simple 1D example: Nested Iterations are transformed from:
+
+    <Iteration x0_blk0; (x_m, x_M, x0_blk0_size)>
+        <Iteration x; (x0_blk0, x0_blk0 + x0_blk0_size - 1, 1)>
+
+    to:
+
+    <Iteration x0_blk0; (x_m, x_M, x0_blk0_size)>
+        <Iteration x; (x0_blk0, MIN(x_M, x0_blk0 + x0_blk0_size - 1)), 1)>
+
```

Comment:
you could now proceed with the example and show that with the new bounds you actually cover the entire iteration space